### PR TITLE
Warn services conflict from multiclusters

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/eds.go [hunk 1]
```diff
@@ -353,23 +353,32 @@ func (s *DiscoveryServer) updateCluster(push *model.PushContext, clusterName str
...
+		pc.Mutex.Lock()
+		ports, ok := pc.ServicePortByHostname[model.Hostname(hostname)]
 		pc.Mutex.Unlock()
+		if ok {
+			if !reflect.DeepEqual(ports, pl) {
+				adsLog.Warnf("service %s within cluster %s does not match the one in primary cluster", hostname, cluster)
+			}
+		} else {
+			adsLog.Debugf("service %s within cluster %s occurs before primary cluster, will only take services within primary cluster", hostname, cluster)
+		}
```
Comment (rshriram) [inline pilot/pkg/proxy/envoy/v2/eds.go]: Isn’t this called for consul, service entries etc ?

Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/eds.go]: no, it is only for k8s

**Pull request merged**